modules/@apostrophecms/doc-type/ui/apos/components/AposDocHistory.vue

```diff
@@ -0,0 +1,144 @@
+<template>
+  <AposModal
+    :modal="modal" :modal-title="modalTitle"
+    @esc="cancel" @no-modal="$emit('safe-close')"
+    @inactive="modal.active = false" @show-modal="modal.showModal = true"
+  >
+    <template #primaryControls>
+      <AposButton
+        type="default" label="Finished"
+        @click="cancel"
+      />
+    </template>
+    <template #main>
+      <AposModalBody>
+        <template #bodyMain>
+          <AposTree
+            :rows="rows" :headers="headers"
+            :icons="icons" :hide-header="true"
+          />
+        </template>
+      </AposModalBody>
+    </template>
+  </AposModal>
+</template>
+
+<script>
+import AposModalParentMixin from 'Modules/@apostrophecms/modal/mixins/AposModalParentMixin';
+import AposTableMixin from 'Modules/@apostrophecms/modal/mixins/AposTableMixin';
+import dayjs from 'dayjs';
+
+export default {
+  name: 'AposDocHistory',
+  mixins: [ AposModalParentMixin, AposTableMixin ],
+  props: {
+    doc: {
+      type: Object,
+      required: true
+    }
+  },
+  emits: [ 'trash', 'search', 'safe-close' ],
+  data() {
+    return {
+      modalTitle: `${this.doc.title} Versions`,
+      modal: {
+        active: false,
+        type: 'slide',
+        showModal: false
+      },
+      versions: [],
+      options: {
+        columns: [
+          {
+            label: 'Edited on',
+            name: 'editedAt',
+            labelIcon: 'calendar'
+          },
+          {
+            label: 'Edited by',
+            name: 'author',
+            labelIcon: 'account-box'
+          }
+        ]
+      }
+    };
+  },
+  computed: {
+    rows() {
+      const rows = [];
+      if (!this.versions || !this.headers.length) {
+        return [];
+      }
+
+      this.versions.forEach(page => {
+        const data = {};
+
+        this.headers.forEach(column => {
+          data[column.name] = page[column.name];
+          data._id = page._id;
+          data.children = page.children;
+        });
+        rows.push(data);
+      });
+
+      return rows;
+    }
+  },
+  async mounted() {
+    this.modal.active = true;
+
+    await this.getVersions();
+  },
+  methods: {
+    async getVersions () {
+      let docVersions = (await apos.http.post(
+        '/api/v1/@apostrophecms/version/list', {
+          busy: true,
+          body: {
+            _id: this.doc._id
+          }
+        }
+      )).versions;
+
+      docVersions = this.formatVersions(docVersions);
+
+      if (docVersions) {
+        this.versions = docVersions;
+      }
+    },
+    openEditor(event) {
```

Comment:
Is this connected to anything? I don't see it being invoked.

Is editing the document an action that makes sense from the doc history modal? If you did, what relationship would that action bear to a particular version...?

I have thoughts on implementation but I think these two questions should be answered first so I know what I'm talkin' bout.

Reply:
No, that was from the component I copied to start this one. I'll remove this method.